### ISSUE TITLE
Problem with tf.train.FloatList

### ISSUE THREAD

**Issue by mcbe1985** [tensorflow/tensorflow#37794]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04):  Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if
the issue happens on mobile device:  /
- TensorFlow installed from (source or
binary): Source, 2.1 - TensorFlow version (use command below): 
- Python version: 3.6.9 - Bazel
version (if compiling from source):
- GCC/Compiler version (if compiling from
source): 
- CUDA/cuDNN version: - GPU model and memory: 10.2

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

I have been trying to convert my Pascal VOC annotations (xml files + jpg) to TFRecords for training and validation datasets but I seem to be running into problems.
My bounding boxes are correct in the xml files (as in they don't go out of bounds) but whenever I start converting them to TFRecords they suddenly change value mid through. I have narrowed the specific code down to one line:
float_list=tf.train.FloatList(value=ymax)
After this line (the same for xmin, xmax and ymin) the values that I've calculated beforehand suddenly change. For example when I have 2 values in my ymax list and then generate the float_list I get the following:
[0.7670886848546281, 0.8025317614591573]
value: 0.7670887112617493
value: 0.8025317788124084

I don't really understand why they change but in some circumstances it makes the value go to 1.0 which makes my loss keep going to nan. 

My question is: is there any way to fix this or am i somehow doing this all wrong? I have provided the conversion script below.

Colab link:
https://colab.research.google.com/drive/1hKXjsgWdKUIAAeQHM5KiRL2iRhxKWSRu

```Python
'''
This program is meant for transforming .xml files to TFRecords.
In the program itself there is no need for changes. The only things
one has to change are the AIBASEMAP and IMAGEMAP locations when 
necessary!

If needed one could also change some settings in the FLAGS, although
this is not deemed necessary if the folder structure stays the same.
'''
import time
import os
from os import listdir
from os.path import isdir, isfile, join

import hashlib

from absl import app, flags, logging
from absl.flags import FLAGS
import tensorflow as tf
import lxml.etree
import tqdm
import time

# Set the basemap from which we're working 
# This is different for each device! Change accordingly!
AIBASEMAP = '/Code/yolov3-tf2-master/'                         #XXX JupyterLab

# Set the image map from which we're working
IMAGEBASEMAP = '/Images/Trash2_updated'


flags.DEFINE_string('data_dir', 'Directory with images and annotations',
                    'Directory with images and annotations')
flags.DEFINE_enum('split', 'train', [
                  'train', 'val'], 'train') #'specify train or val spit')
flags.DEFINE_string('output_file_train', 'Path to output train.tfrecord', 'Path to output train.tfrecord')
flags.DEFINE_string('output_file_val', 'Path to output val.tfrecord', 'Path to output val.tfrecord')
flags.DEFINE_string('classes', 'Path to .names file with labels', 'Class names record')

def build_example(annotation, class_map, path, i):
    check = True
    img_path = path + '.jpg'
    img_raw = open(img_path, 'rb').read()
    key = hashlib.sha256(img_raw).hexdigest()

    width = int(annotation['size']['width'])
    height = int(annotation['size']['height'])

    xmin = []
    ymin = []
    xmax = []
    ymax = []
    classes = []
    classes_text = []
    truncated = []
    views = []
    difficult_obj = []
    if 'object' in annotation:
        for obj in annotation['object']:
            difficult = bool(int(obj['difficult']))
            difficult_obj.append(int(difficult))

            xmin.append(float(obj['bndbox']['xmin']) / width)
            ymin.append(float(obj['bndbox']['ymin']) / height)
            xmax.append(float(obj['bndbox']['xmax']) / width)
            ymax.append(float(obj['bndbox']['ymax']) / height)
            classes_text.append(obj['name'].encode('utf8'))
            classes.append(class_map[obj['name']])
            truncated.append(int(obj['truncated']))
            views.append(obj['pose'].encode('utf8'))

    example = tf.train.Example(features=tf.train.Features(feature={
        'image/height': tf.train.Feature(int64_list=tf.train.Int64List(value=[height])),
        'image/width': tf.train.Feature(int64_list=tf.train.Int64List(value=[width])),
        'image/filename': tf.train.Feature(bytes_list=tf.train.BytesList(value=[
            annotation['filename'].encode('utf8')])),
        'image/source_id': tf.train.Feature(bytes_list=tf.train.BytesList(value=[
            annotation['filename'].encode('utf8')])),
        'image/key/sha256': tf.train.Feature(bytes_list=tf.train.BytesList(value=[key.encode('utf8')])),
        'image/encoded': tf.train.Feature(bytes_list=tf.train.BytesList(value=[img_raw])),
        'image/format': tf.train.Feature(bytes_list=tf.train.BytesList(value=['jpeg'.encode('utf8')])),
        'image/object/bbox/xmin': tf.train.Feature(float_list=tf.train.FloatList(value=xmin)),
        'image/object/bbox/xmax': tf.train.Feature(float_list=tf.train.FloatList(value=xmax)),
        'image/object/bbox/ymin': tf.train.Feature(float_list=tf.train.FloatList(value=ymin)),
        'image/object/bbox/ymax': tf.train.Feature(float_list=tf.train.FloatList(value=ymax)),
        'image/object/class/text': tf.train.Feature(bytes_list=tf.train.BytesList(value=classes_text)),
        'image/object/class/label': tf.train.Feature(int64_list=tf.train.Int64List(value=classes)),
        'image/object/difficult': tf.train.Feature(int64_list=tf.train.Int64List(value=difficult_obj)),
        'image/object/truncated': tf.train.Feature(int64_list=tf.train.Int64List(value=truncated)),
        'image/object/view': tf.train.Feature(bytes_list=tf.train.BytesList(value=views)),
    }))
    return example


def parse_xml(xml):
    if not len(xml):
        return {xml.tag: xml.text}
    result = {}
    for child in xml:
        child_result = parse_xml(child)
        if child.tag != 'object':
            result[child.tag] = child_result[child.tag]
        else:
            if child.tag not in result:
                result[child.tag] = []
            result[child.tag].append(child_result[child.tag])
    return {xml.tag: result}

def conversion(image_list, writer, class_map):
    i = 0
    for image in tqdm.tqdm(image_list):
        i += 1
        temp = image
        try:
            image = image.split('.')[0]
            path_name = image.replace('.', '')
            annotation_xml = path_name + '.xml'
            annotation_xml = lxml.etree.fromstring(open(annotation_xml, 'rb').read())
            annotation = parse_xml(annotation_xml)['annotation']
            tf_example = build_example(annotation, class_map, path_name, i)
            writer.write(tf_example.SerializeToString())
        except Exception as exception:
            print('Er is een fout bij: ', path_name)
            with open('exception.log', '+w') as error_log_file:
                error_log_file.write(str(exception))
            quit()
            # check for XML syntax errors
        except etree.XMLSyntaxError as err:
            print('XML Syntax Error, see error_syntax.log, Error in: ', path_name)
            with open('error_syntax.log', '+a') as error_log_file:
                error_log_file.write('\nEr is een Fout in: ', path_name)
                error_log_file.write(str(err.error_log))
            quit()


def main(_argv):
    class_map = {name: idx for idx, name in enumerate(
        open(FLAGS.classes).read().splitlines())}
    logging.info("Class mapping loaded: %s", class_map)

    directories = [f for f in listdir(FLAGS.data_dir) if isdir(join(FLAGS.data_dir, f))]
    print('Start trainingsdata conversion.')
    writer = tf.io.TFRecordWriter(FLAGS.output_file_train)
    image_list = open(os.path.join(
        FLAGS.data_dir, 'Trash', 'solutions_%s.txt' % FLAGS.split)).read().splitlines()
    logging.info("Image list loaded: %d", len(image_list))
    conversion(image_list, writer, class_map)
    writer.close()
    print('Done')
    print('Start validationdata conversion.')
    writer = tf.io.TFRecordWriter(FLAGS.output_file_val)
    image_list = open(os.path.join(
        FLAGS.data_dir, 'Trash', 'validation_data.txt' )).read().splitlines()
    logging.info("Image list loaded: %d", len(image_list))
    conversion(image_list, writer, class_map)
    writer.close()
    logging.info("Done")


if __name__ == '__main__':
    app.run(main)
```

**Comment by ashutosh1919** [comment]
@mcbe1985 , Please provide colab link here. The code given is not structured and well indented.

**Comment by mcbe1985** [comment]
The colab link is; https://colab.research.google.com/drive/1hKXjsgWdKUIAAeQHM5KiRL2iRhxKWSRu

I've added the colab to the opening post as well.

**Comment by gowthamkpr** [comment]
@mcbe1985 Is it still an issue or did you resolve it. Thanks!

**Comment by mcbe1985** [comment]
Fixed it during my research, was a mistake in the annotations which made it do weird stuff.

**Comment by kaka-lin** [comment]
@mcbe1985 How do you solved it?

I have the same issue

**Comment by mcbe1985** [comment]
> @mcbe1985 How do you solved it?
> 
> I have the same issue

In most cases there's a mistake in your annotations (as was the case here). For the github issue and repo where I figured it out check https://github.com/zzh8829/yolov3-tf2/issues/214

Hope that helps.

**Comment by kaka-lin** [comment]
Hi @mcbe1985,

Sorry I am not describe clear my issue.
I already open a new issue #42088
